molgenis-security/src/main/java/org/molgenis/security/audit/AuditSettingsRepositoryDecorator.java

```diff
@@ -0,0 +1,59 @@
+package org.molgenis.security.audit;
+
+import java.util.HashMap;
+import java.util.stream.Stream;
+import org.molgenis.audit.AuditEvent;
+import org.molgenis.audit.AuditEventPublisher;
+import org.molgenis.data.AbstractRepositoryDecorator;
+import org.molgenis.data.Entity;
+import org.molgenis.data.Repository;
+import org.molgenis.data.security.audit.AuthenticationUtils;
+
+/** Publishes {@link AuditEvent}s when changes are made to {@link AuditSettings}. */
+public class AuditSettingsRepositoryDecorator extends AbstractRepositoryDecorator<Entity> {
+
+  static final String AUDIT_SETTING_CHANGED = "AUDIT_SETTING_CHANGED";
```

Comment:
you've made it (almost) so generic that it can actually be re-used for all settings repositories, I think(!)

Reply:
as discussed, will do this in another PR if possible

Reply:
did it anyway!